components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java

```diff
@@ -0,0 +1,1667 @@
+                        validationResponse,
+                        apiDefinition,
+                        asyncApiDocument.asyncapi,
+                        asyncApiDocument.info.title,
+                        asyncApiDocument.info.version,
+                        null,
+                        asyncApiDocument.info.description,
+                        null
+                );
+            }
+
+            validationResponse.setParser(this);
+            if (returnJsonContent) {
+                validationResponse.setJsonContent(apiDefinition);
+            }
+        } else {
+            if (validationErrorMessages != null){
+                validationResponse.setValid(false);
+                for (String errorMessage: validationErrorMessages){
+                    AsyncApiParserUtil.addErrorToValidationResponse(validationResponse, errorMessage);
+                }
+            }
+        }
+        return validationResponse;
+    }
+
+    @Override
+    public String populateCustomManagementInfo(String oasDefinition, SwaggerData swaggerData) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String getOASDefinitionForStore(API api, String oasDefinition, Map<String, String> hostsWithSchemes) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String getOASDefinitionForStore(APIProduct product, String oasDefinition, Map<String, String> hostsWithSchemes) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String getOASDefinitionForPublisher(API api, String oasDefinition) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String getOASVersion(String oasDefinition) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String getOASDefinitionWithTierContentAwareProperty(String oasDefinition, List<String> contentAwareTiersList, String apiLevelTier) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String processOtherSchemeScopes(String resourceConfigsJSON) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public API setExtensionsToAPI(String swaggerContent, API api) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String processDisableSecurityExtension(String swaggerContent) throws APIManagementException{
+        return null;
+    }
+
+    @Override
+    public String injectMgwThrottlingExtensionsToDefault(String swaggerContent) throws APIManagementException{
+        return null;
+    }
+
+    public String generateAsyncAPIDefinition(API api) throws APIManagementException {
+        Aai20Document aaiDocument = new Aai20Document();
+        aaiDocument.info = aaiDocument.createInfo();
+        aaiDocument.info.title = api.getId().getName();
+        aaiDocument.info.version = api.getId().getVersion();
+        if (api.getType() != "WEBSUB") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.